### ISSUE TITLE
Wildcard Emission

### ISSUE THREAD

**Issue by yooneskh** [sindresorhus/emittery#80]
Hi, thanks for your great work.
Does Emittery support wildcard emit?

**Comment by sindresorhus** [comment]
No. I assume you mean `emit('namespace.*')` and not `emit('*')`?

**Comment by yooneskh** [comment]
Yeah, and even more general form of it as `emit('x.*.y')`. Is there any plan to support it? @sindresorhus

**Comment by sindresorhus** [comment]
It's not something I plan to work on, but I'm open to adding it if someone does a **good** pull request. It's blocked by https://github.com/sindresorhus/emittery/issues/31 though.

**Comment by sindresorhus** [comment]
Closing this. Wildcard emit is architecturally problematic: it turns emit from an O(1) map lookup into an O(n) scan of all registered event names, and it breaks type safety since each matched event may expect different data shapes. No major event library supports wildcard emit (even EventEmitter2 only supports wildcards on the subscribe side).

The use cases are already covered:

```js
// Listen to multiple specific events
emitter.on(["mouse.click", "mouse.move", "mouse.scroll"], handler);

// Pattern matching with onAny()
emitter.onAny(event => {
    if (typeof event.name === "string" && event.name.startsWith("mouse.")) {
        handleMouseEvent(event);
    }
});
```

If you need full wildcard/namespace support, EventEmitter2 is a good fit for that.
